velox/connectors/hive/SplitReader.cpp

```diff
@@ -364,6 +377,10 @@ std::vector<TypePtr> SplitReader::adaptColumns(
           1,
           connectorQueryCtx_->memoryPool());
       childSpec->setConstantValue(constant);
+    } else if (
+        rowIndexColumn != nullptr && fieldName == rowIndexColumn->name()) {
+      childSpec->setConstantValue(BaseVector::createNullConstant(
```

Comment:
Why do we supply null constant for it?  It might cause issue in other places (e.g. breaking vector reuse)  because the actual values in result vector are not null.  You can add extra flags to `ScanSpec` to indicate the special status of this node.

Reply:
Thanks, I added new flag ``isExplicitRowNumber  for this.